fluent-syntax/src/index.ts

```diff
@@ -2,11 +2,11 @@ import { Resource } from "./ast.js";
 import { FluentParser, FluentParserOptions } from "./parser.js";
 import { FluentSerializer, FluentSerializerOptions } from "./serializer.js";
 
-export * from "./ast";
-export * from "./errors";
-export * from "./parser";
-export * from "./serializer";
-export * from "./visitor";
+export * from "./ast.js";
+export * from "./errors.js";
+export * from "./parser.js";
+export * from "./serializer.js";
+export * from "./visitor.js";
```

Comment:
Why does it import `.js` files from `.ts` file?

Reply:
Because this is the cleanest way of producing valid ES module code in the `esm/` directories. Node.js resolution requires the `.js` extension for `import` calls, and TypeScript automatically resolves e.g. `./ast.js` as `./ast.ts`.

Without these, we'd need a separate build step after the `tsc` call that would modify all of the relative paths to use fully qualified extensions. It'd be nice if TypeScript supported that natively, but, well, it doesn't. This is also why we need the `esm/package.json` files, as TypeScript _only_ handles the `.js` extension like this, and not e.g. `.mjs`.